### PR TITLE
Bug 1790407: [3.11] Fix reinitialization of deny-all NetworkPolicy state on restart

### PR DESCRIPTION
**Pull request by fbac**
backport of https://github.com/openshift/sdn/pull/105/commits/01789d2e93641ca5b23c0448aacd196acc9d71a8 and https://github.com/openshift/sdn/pull/105/commits/2da9083753e41e255f4e3e03f72274997b750d5c

/cc @juanluisvaladas

### CODE REVIEW

**Comment by juanluisvaladas** [comment]
Hey Borja, thanks for the PR.

Tests are failing because you are assuming 0 is always present, which isn't true: https://github.com/openshift/sdn/commit/2da9083753e41e255f4e3e03f72274997b750d5c

File: pkg/network/node/ovscontroller_test.go [hunk 1]
```diff
@@ -968,6 +991,79 @@ func TestFindUnusedVNIDs(t *testing.T) {
 	}
 }
 
+func TestFindPolicyVNIDs(t *testing.T) {
+	testcases := []struct {
+		flows       []string
+		policyVNIDs sets.Int
+	}{
+		{
```
Comment (juanluisvaladas) [inline pkg/network/node/ovscontroller_test.go]: This test is wrong:

https://github.com/openshift/sdn/pull/103/commits/f5c2488a9475cca70e2a3bad76add6915ef54726#diff-8f7a8126747765bebab51b916de23c99R1000-R1005

Comment (juanluisvaladas) [reply pkg/network/node/ovscontroller_test.go]: Looks like I'm highlighting the wrong line. I meant this test:
`
			// 0 must always be present
			flows: []string{
				"table=80,priority=0 actions=drop",
			},
			policyVNIDs: sets.NewInt(0x0),
		},`

**Comment by juanluisvaladas** [comment]
Ah I see where the conflict is coming from, the cherry-pick is in the wrong order.
First you need to do  the cherry pick of openshift/sdn@01789d2 and after that of openshift/sdn@2da9083 . You did it the other way around

**Comment by juanluisvaladas** [comment]
/test unit

Odd, the unit test works on my machine:

$ gh pr checkout 24620
From github.com:openshift/origin
 * [new ref]               refs/pull/24620/head -> sdn-backport-3.11-bz1793952
Updating files: 100% (36391/36391), done.
Switched to branch 'sdn-backport-3.11-bz1793952'
$ go test github.com/openshift/origin/pkg/network/node
ok  	github.com/openshift/origin/pkg/network/node	0.050s

Is this a legit failure?

**Comment by juanluisvaladas** [comment]
@fbac can you please rebase this?

**Comment by danwinship** [comment]
> can you please rebase this? I'm pretty sure you didn't introduce the issue so I guess that would fix it.

@juanluisvaladas FTR the tests are run by merging the PR into master first, so rebasing shouldn't actually fix test failures. (The failures were because 3.11 CI was broken for a few weeks because of CI system migration and people not caring enough about 3.11.)

**Comment by danwinship** [comment]
/hold
to stop the retests; 3.11 e2e-gcp looks broken...

**Comment by juanluisvaladas** [comment]
/hold cancel
The e2e-gcp has passed so there is no reason to hold this any longer

**Pull request merged**